app/src/main/java/com/github/se/icebreakrr/model/tags/TagsRepository.kt

```diff
@@ -0,0 +1,250 @@
+package com.github.se.icebreakrr.model.tags
+
+import android.util.Log
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.QuerySnapshot
+
+class TagsRepository(private val db: FirebaseFirestore) {
+  private val collectionPath = "Tags"
+
+  /**
+   * Function that get all the tags present in the firestore database
+   *
+   * @param onSuccess : when the operation is successful, returns the list of TagsCategory got from
+   *   the database
+   * @param onFailure : When the operation fails, log an error and returns the onFailure with the
+   *   erorr
+   */
+  fun getAllTags(onSuccess: (List<TagsCategory>) -> Unit, onFailure: (Exception) -> Unit) {
+    val categories: MutableList<TagsCategory> = mutableListOf()
+    db.collection(collectionPath)
+        .get()
+        .addOnFailureListener { e: Exception ->
+          Log.e("TagsRepository", "[getAllTags] Could not get the tags : $e")
+          onFailure(e)
+        }
+        .addOnSuccessListener { docs: QuerySnapshot ->
+          for (doc in docs.documents) {
+            if (doc.exists()) {
+              firestoreToTags(
+                  doc,
+                  { categories.add(it) },
+                  {
+                    Log.e(
+                        "TagsRepository",
+                        "[getAllTags] Could not fetch firebase to get the tags : $it")
+                    onFailure(it)
+                  })
+            }
+          }
+          onSuccess(categories)
+        }
+  }
+
+  /**
+   * Function to add a tag to firebase in a category
+   *
+   * @param onFailure : could happen when the document is retrieved, when we convert document into
+   *   TagsCategory, when the tag already exists in this category or if the category doesn't exist
+   * @param category : category in which you want to add a tag
+   * @param name : name of the new tag that you want to add
+   */
+  fun addTag(onFailure: (Exception) -> Unit, category: CategoryString, name: String) {
+    val docRef = db.collection(collectionPath).document(category.name)
+    docRef
+        .get()
+        .addOnFailureListener { exception: Exception ->
+          Log.e("TagsRepository", "[addTag] Could not retrieve the document: $exception")
+          onFailure(exception)
+        }
+        .addOnSuccessListener { documentSnapshot: DocumentSnapshot ->
+          if (documentSnapshot.exists()) {
+            var tagsCategory = TagsCategory("", "#FFFFFFFF", listOf())
```

Comment:
Maybe having a by default constructor can be useful to make the code lighter and more readable, since those default values aren't used and are overwritten afterwards